pkg/cover/canonicalizer.go

```diff
@@ -71,9 +77,11 @@ func (can *Canonicalizer) NewInstance(modules []host.KernelModule) *Canonicalize
 	instToCanonicalMap := make(map[uint32]*canonicalizerModule)
 	canonicalToInstMap := make(map[uint32]*canonicalizerModule)
 	for _, module := range modules {
+		discard := false
 		canonicalModule, found := can.modules[module.Name]
 		if !found || canonicalModule.Size != module.Size {
-			log.Fatalf("kernel build has changed; instance module %v differs from canonical", module.Name)
+			log.Errorf("kernel build has changed; instance module %v differs from canonical", module.Name)
+			discard = true
```

Comment:
Now that you continue execution from here, won't it crash on `canonicalModule.Addr` below if `found` is `false`?
If it crashes, please adjusts tests to cover this.

Reply:
When `found` is `false` canonicalModule resorts to its default values (https://go.dev/blog/maps). In this case, canonicalModule.Addr defaults to 0. The new `TestChangingModules` test does cover this  case